File: forge/src/main/java/com/example/examplemod/platform/ForgePlatformHelper.java
package com.example.examplemod.platform;

import com.example.examplemod.platform.services.IPlatformHelper;
import net.minecraftforge.fml.loading.FMLLoader;
import net.minecraftforge.fml.loading.LoadingModList;

public class ForgePlatformHelper implements IPlatformHelper {

    @Override
    public String getPlatformName() {

        return "Forge";
    }

    @Override
    public boolean isModLoaded(String modId) {

        return LoadingModList.getModFileById(modId) != null;
    }

    @Override
    public boolean isDevelopmentEnvironment() {

        return !FMLLoader.isProduction();
    }
}

File: MultiLoader-Template-with-kotlin-and-kotlin-dsl/Forge/src/main/kotlin/com/example/examplemod/platform/ForgePlatformHelper.kt
package com.example.examplemod.platform

import com.example.examplemod.platform.services.IPlatformHelper
import net.minecraftforge.fml.ModList
import net.minecraftforge.fml.loading.FMLLoader

class ForgePlatformHelper : IPlatformHelper {
    override fun getPlatformName(): String {
        return "Forge"
    }

    override fun isModLoaded(modId: String?): Boolean {
        return ModList.get().isLoaded(modId)
    }

    override fun isDevelopmentEnvironment(): Boolean {
        return !FMLLoader.isProduction()
    }
}